sql/analyzer/hoist_select_exists.go

```diff
@@ -35,12 +37,30 @@ func hoistSelectExists(
 	scope *Scope,
 	sel RuleSelector,
 ) (sql.Node, transform.TreeIdentity, error) {
+	a.LogNode(n)
+	nodeAliases, err := getTableAliases(n, scope)
```

Comment:
we should only lazily build nodeAliases, otherwise we're slowing down unrelated queries

Reply:
Reorganized the code to get aliases only if we're about to hoist a subquery.